### PR TITLE
feat(WebSocketModel): add async methods and call fetch on error or close

### PR DESCRIPTION
**Pull request by MariusPisto**

### CODE REVIEW

File: Model/WebSocketModel.js [hunk 1]
```diff
@@ -6,11 +6,12 @@ export class WebSocketModel {
...
         if (WebSocketModel.#O_singleton === null) {
             WebSocketModel.#O_singleton = this;
-            this.#O_websocket = new WebSocket(O_WEB_SOCKET_CONSTANTS.url)
+            this.#O_websocket = new WebSocket(O_WEB_SOCKET_CONSTANTS.url);
+            this.#O_websocket.onopen = () => {
+                this.#O_websocket.send("Hello");
+            }
         }
 
         console.log(this.#O_websocket);
```
Comment (GuillaumeKusiak) [inline Model/WebSocketModel.js:17]: it's necessary ?

File: Service/DataService.js [hunk 2]
```diff
@@ -28,49 +29,68 @@ export class DataService {
...
+        this.#I_connectionStatus = I_connectionStatus;
     }
 
-    #onMessageWebSocketCallBack (event) {
-        console.log(event);
-        this.#O_singleton.#O_realtimeObservable.notify(event);
-        this.#O_singleton.#O_historyObservable.notify(event);
+    #listenWebSocket() {
+        this.#O_websocketModel.onWebSocketMessage((event) => {
+            console.log(event);
```
Comment (GuillaumeKusiak) [inline Service/DataService.js:38]: same here ?

Comment (MattiasGervilliers) [inline Service/DataService.js:38]: remove console.log ?

File: Service/DataService.js [hunk 3]
```diff
@@ -28,49 +29,68 @@ export class DataService {
...
+            this.#O_realtimeObservable.notify(event);
+            this.#O_historyObservable.notify(event);
+        });
+        this.#O_websocketModel.onWebSocketClose((event) => {
+            console.log(event);
```
Comment (MattiasGervilliers) [inline Service/DataService.js:43]: same here

Comment (GuillaumeKusiak) [inline Service/DataService.js:43]: same here also

**Review by MattiasGervilliers — CHANGES_REQUESTED** [review]

**Review by GuillaumeKusiak — APPROVED** [review]

File: Service/DataService.js (hunk 2 above)
Comment (MariusPisto) [reply Service/DataService.js:38]: not during development phase

**Review by MattiasGervilliers — APPROVED** [review]

**Pull request merged**